core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCache.java

```diff
@@ -435,6 +450,39 @@ public T retry() {
     return zr.retry();
   }
 
+  /*
+   * Some of the table configuration parameters are possibly not in ZooKeeper yet and even though
+   * you are able you should not put a watcher on them to observe the NodeCreated event. If you do
+   * that and the ZNode for that configuration is never created, you will have no way to delete the
+   * watcher you created for it in Zookeeper. Unremovable watchers that are created for for table
+   * configs that never actually have a Znode is causing memory leaks in Accumulo instances that
+   * create and delete tables at a high rate.
+   */
+
+  private boolean isWatched(ZooKeeper zooKeeper, String zPath)
+      throws KeeperException, InterruptedException {
+
+    boolean watched = true;
+
+    for (String possiblyWatchedConfig : extraConfigs) {
+      Stat stat;
+      if (zPath.endsWith(possiblyWatchedConfig)) {
+        try {
+          stat = zooKeeper.exists(zPath, false);
```

Comment:
make a comment that this will remove the watcher?

Reply:
It doesn't remove watcher.  It checks for node existence with placing a watch on the node.  We are only checking or existence and nothing else here.